src/terminal/parser/stateMachine.hpp

```diff
@@ -147,6 +147,15 @@ namespace Microsoft::Console::VirtualTerminal
         bool _isInAnsiMode;
 
         std::wstring_view _run;
+        std::wstring_view _currentString;
+
+        size_t _runOffset;
+        size_t _runSize;
+
+        void _ActuateCurrentRun()

```

Comment:
In urgent need of better naming & meaningful comments.

Reply:
At least warn people that you need to call `_ActuateCurrentRun` before actually using `_run`